src/main/java/com/advent/of/code/day7/CrabAdjuster.java

```diff
@@ -0,0 +1,38 @@
+package com.advent.of.code.day7;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import static java.util.Comparator.comparingInt;
+
+class CrabAdjuster {
+    int calculateAdjustmentCostForFixedBurnRate(List<Integer> positions) {
+        return calculateCost(positions, (currentPosition, positionCandidate) -> Math.abs(currentPosition - positionCandidate));
+    }
+
+    int calculateAdjustmentCostForIncreasingBurnRate(List<Integer> positions) {
+        return calculateCost(positions, (currentPosition, positionCandidate) -> {
+            int n = Math.abs(currentPosition - positionCandidate);
+            return ((1 + n) * n) / 2;
+        });
+    }
+
+    private int calculateCost(List<Integer> positions, BiFunction<Integer, Integer, Integer> currentCostCalculator) {
+        final int minPosition = positions.stream().min(comparingInt(Integer::intValue)).get();
+        final int maxPosition = positions.stream().max(comparingInt(Integer::intValue)).get();
```

Comment:
Super zapis! Ja do tego specjalnie napisałem metodę, a widzę, że można to załatwić jednym zgrabnym użyciem streama. Przy następnej okazji spróbuję sam użyć takiej konstrukcji :)

Reply:
tu tylko trzeba uważać na tego `get()` - o czym IntelliJ informuje ;P chodzi o to, że wynik min/max to `Optional` czyli taki wrapper class na wartość która jest ale może jej nie być - to tylko zadanie, w którym widzę input więc mam pewność, że ten min/max jakiś będzie ale w takim kodzie produkcyjnym, gdzie user wrzuca wejścia jakie mu się podobają, to by nie przeszło bo by wywaliło błąd prędzej czy później - a obsługa tego błędu już wymaga użycia jakiegoś `orElse()`, `orElseGet()` albo `orElseThrow()` zamiast `get()`